build-test.cmd

```diff
@@ -6,8 +6,17 @@ set __BuildArch=x64
 set __VCBuildArch=x86_amd64
 set __BuildType=Debug
 set __BuildOS=Windows_NT
-set __VSVersion=vs2015
-set __VSToolsRoot=%VS140COMNTOOLS%
+
+:: Default to highest Visual Studio version available
+if defined VS150COMNTOOLS (
```

Comment:
I think we should not build the policy to pickup the latest version by default but rather have the user specify it. I think we should introduce a command line argument to specify picking up VS2017 so that there is no guesswork around which VS got used for building the repo. 

The default is always VS 2015 and VS 2017 should be opt-in.